lib/tests/test_working_copy_sparse.rs

```diff
@@ -181,7 +181,10 @@ fn test_sparse_commit() {
     // Set sparse patterns to also include dir2/
     let mut locked_wc = wc.start_mutation();
     let sparse_patterns = vec![dir1_path, dir2_path];
-    locked_wc.set_sparse_patterns(sparse_patterns).unwrap();
+    // XXX: set_sparse_patterns() fails because it tries to overwrite "dir2/file1"
+    // XXX: what's the expected behavior?
+    assert!(locked_wc.set_sparse_patterns(sparse_patterns).is_err());
+    /*
     locked_wc.finish(repo.op_id().clone());
```

Comment:
@martinvonz what's expected here?

`set_sparse_patterns()` fails because newly included "dir2/file1" already exists in the working directory. Some options are:

a. rewrite `test_sparse_commit()` to not have un-ignored files
b. make `set_sparse_patterns()` ignore EEXIST silently (and leave working copy dirty?)

Reply:
I had been thinking that we'd eventually do (b). Do you agree? I'm fine with doing just (a) for now if you prefer.